packages/yoroi-extension/app/api/ada/transactions/shelley/ledgerTx.js

```diff
@@ -161,14 +161,42 @@ function toLedgerTokenBundle(
         assetNameHex: Buffer.from(assetName.name()).toString('hex'),
       });
     }
+    // sort by asset name to the order specified by rfc7049
+    tokens.sort(
+      (token1, token2) => compareCborKey(token1.assetNameHex, token2.assetNameHex)
+    );
     assetGroup.push({
       policyIdHex: Buffer.from(policyId.to_bytes()).toString('hex'),
       tokens,
     });
   }
+  // sort by policy id to the order specified by rfc7049
+  assetGroup.sort(
+    (asset1, asset2) => compareCborKey(asset1.policyIdHex, asset2.policyIdHex)
+  );
   return assetGroup;
 }
 
+function compareCborKey(hex1: string, hex2: string): number {
+  if (hex1.length < hex2.length) {
+    return -1;
+  }
+  if (hex2.length < hex1.length) {
+    return 1;
+  }
+  const bytes1 = [...Buffer.from(hex1, 'hex')];
+  const bytes2 = [...Buffer.from(hex2, 'hex')];
+  for (let i = 0; i < bytes1.length; i++) {
+    if (bytes1[i] < bytes2[i]) {
+      return -1;
+    }
+    if (bytes2[i] < bytes1[i]) {
+      return 1;
+    }
+  }
+  return 0;
+}
```

Comment:
I don't think you need to parse HEX here to compare the values, HEX pairs are always alphabetically ordered for higher bytes correctly so strings can be just compared directly when we know they are of the same length.

Can be something like:
```
const cmp = (a, b) = a === b ? 0 : (a < b ? -1 : 1);
return cmp(a.length, b.length) || cmp(a, b)
```

But not critical

Reply:
I think it would be good to avoid parsing the strings and copying them into arrays within the comparing function, because it will be executed many times for the same values of key1 and key2 when sorting many assets